integration/testrunner/grpc.go

```diff
@@ -307,29 +353,45 @@ func createConnection(ctx context.Context, t *testing.T, addr, idToken string) *
 // This query is used to find the relevant audit log in BigQuery, which we assume will be added by the server.
 // We specifically look up the log using the UUID specified in the request as we know the server will add that
 // as the resource name, and provides us a unique key to find logs with.
-func (g *GRPC) makeQueryForGRPCUnary(id string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE jsonPayload.resource_name=?", g.ProjectID, g.DatasetQuery)
-	if g.RequireJustification {
-		// TODO(#265): For back-compatibility, we need to ensure justification in either metadata key.
-		queryString += ` AND jsonPayload.metadata.justification IS NOT NULL`
+func (g *GRPC) makeQueryForGRPCUnary(id string, fieldsNameMap [][]string) *bigquery.Query {
+	queryString := "SELECT "
+	for _, v := range fieldsNameMap {
+		queryString += fmt.Sprintf("%s as %s,", v[0], v[1])
 	}
-	queryString += ` AND jsonPayload.service_name IS NOT NULL`
-	queryString += ` AND jsonPayload.method_name IS NOT NULL`
-	queryString += ` AND jsonPayload.authentication_info.principal_email IS NOT NULL`
-	queryString += ") AS INT64)"
+	queryString += fmt.Sprintf("FROM `%s.%s` WHERE jsonPayload.resource_name='%s'", g.ProjectID, g.DatasetQuery, id)
 	return makeQuery(*g.BigQueryClient, id, queryString)
 }
 
+func parseQueryResultForGRPCUnary(tb testing.TB, value []bigquery.Value) GRPCFields {
+	tb.Helper()
+	result := GRPCFields{}
+	elem := reflect.ValueOf(&result).Elem()
```

Comment:
I like reflection, but my #1 rule of reflection is that all reflection code must have at least twice as many documentation lines as code lines 😄

Reply:
Updated.

Reply:
In this case, each value in `[]bigquery.Value` would have type `string`. But if we directly use `elem.Field(i).SetString(v)`, there will be a error saying that `cannot use v (variable of type "cloud.google.com/go/bigquery".Value) as string value in argument to elem.Field(i).SetString: need type assertion`. So `v.(string)` is just for type assertion.